src/client/dfs/dfs.c

```diff
@@ -1577,7 +1598,27 @@ dfs_mkdir(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
 
```

Comment:
If the insert_entry call returns an error, do we leak new_dir.oh?

Since insert_entry is essentially one conditional update, it appears a TX is unnecessary?

Reply:
hmm yes you're right. the change could cause a leaked handle since we could be using conditional update.
yea maybe we don't need a tx at all here. let me remove that in the create case.